### DISCUSSION TITLE
Can not burn Arduino bootloader for STM32F072C8

### DISCUSSION THREAD

**Discussion by 5ghub** [stm32duino/Arduino_Core_STM32#2093] [General]
I have a USB hardware board with STM32F072C8 and I am trying to use the stm32duino to be able to use the board with Arduino IDE. 
The problem I am facing is that I cannot burn the bootloader to the board.

I connect the STM32 BOOT pin to active HIGH (3.3v) and this put my USB in BOOT mode. (I am assumung no other PIN needs to be connected)

It can enumerates successfully in Windows Device Manage as DFU Device. Screenshot below

![Capture](https://github.com/stm32duino/Arduino_Core_STM32/assets/51192937/c0eb2260-4add-4a45-98a1-8679cad39b71)

I also launched CubeProgrammer and despite it can detect the PID and VID correctly, it can not connect to the USB board and dispalys
"Error: Establishing connection with device failed"

See screenshot below

![capture1](https://github.com/stm32duino/Arduino_Core_STM32/assets/51192937/e21d00ed-4471-4bd0-a258-fe163748341d)

I launch Arduino IDE and set everything correctly. However, when I choose "Burn Bootloader", I receive the error "Missing Programmer"
see this screenshot

![Capture3](https://github.com/stm32duino/Arduino_Core_STM32/assets/51192937/32dabe1f-d9c3-4486-b48b-0568bf5198a7)

However, I wrote a simple Arduino sketch that toggles LEDs on the board, I chose Arduino->Sketch->Verify/Compile. Then I took the *.hex file and flashed it to the USB board using J-Link and SWD.  The sketch working on the board successfully and LEDs are toggled as expected.

I am thinking if I can flash the bootloader the same way using J-link and SWD but I do not know how to generate the *.hex file for the bootloader.

Any help in the above issues are much appreciated and would be helpful.   Feel free to ping me on skype: hossamfattah  and any sincere help will be rewarded.

**Comment by fpistm** [comment]
Burn bootloader from Arduino is not implemented as we do not need it. About DFU if the cube programmer is not able to connect it means you have an issue with your hardware. Normally if you choose dfu in the upload method and the board is in the built in bootloader in dfu mode it should flash properly.

**Reply by 5ghub** [reply]
Thanks Frederic for your help.

Can you please clarify what hardware issue might be there? 
If the device enumerates in the Windows device manager successfully, is not that a sign the USB and hardware connections are correct, and right ?

To put the device in DFU Mode, only BOOT pin needs to be HIGH? is that right or is there any other pin needs to be asserted with that?

I am attaching the PCB schematic here. The boards itself working successfully with the following:

1- SWD
2- I am able to toggle GPIOs LEDs on PB13 & PB14 and LEDs are toggling 
3- Also, USB port (PA11 & PA12) and Serial USART (PA2 & PA3) are working and can print and accept characters

can you help what is the issue exactly?
![STM32F072](https://github.com/stm32duino/Arduino_Core_STM32/assets/51192937/c4c5fbf8-c83f-4171-a34a-d0572081897a)


![STM32_01](https://github.com/stm32duino/Arduino_Core_STM32/assets/51192937/71fecc83-11f9-4064-9a48-6a8fe20a4a2f)

Thanks a lot for your great help

Best
Hossam Fattah

**Reply by fpistm** [reply]
You can see the pattern to apply to enter in DFU mode:
https://www.st.com/resource/en/application_note/an2606-stm32-microcontroller-system-memory-boot-mode-stmicroelectronics.pdf

It is the pattern2:
Pattern 2 Boot0(pin) = 1 and nBoot1(bit) = 1

Seems you have a circuitry to select USB. Ensure your USB_SEL signal is at the correct value.

While you can't achieve to connect with STM32CubeProgrammer no need to test Arduino as it is not linked to it.

**Reply by 5ghub** [reply]
Hello Fredrick,

I was able to solve this problem. the issue was in the Windows Device manager showing up as "STM32 DFU device". When I update the driver, it shows up as "STM32 BOOTLOADER" and this fixed the problem and CubeProgammer was able to connect to the board.

Thanks for your help.

**Reply by fpistm** [reply]
Great. Thanks for the feedback.
